firefox_puppeteer/api/places.py

```diff
@@ -0,0 +1,161 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette.errors import TimeoutException
+
+from ..base import BaseLib
+
+
+class Places(BaseLib):
+    """Low-level access to several bookmark and history related actions."""
+
+    bookmark_folders = {'placesRoot': 1,
+                        'bookmarksMenuFolder': 2,
+                        'toolbarFolder': 3,
+                        'tagsFolder': 4,
+                        'unfiledBookmarksFolder': 5,
+                        }
+
+    # Bookmark related helpers #
+
+    def is_bookmarked(self, url):
+        """Checks if the given URL is bookmarked.
+
+        :param url: The URL to Check
+
+        :returns: True, if the URL has been found
+        """
+        return self.marionette.execute_script("""
+          let url = arguments[0];
+
+          let bs = Cc["@mozilla.org/browser/nav-bookmarks-service;1"]
+                   .getService(Ci.nsINavBookmarksService);
+          let ios = Cc["@mozilla.org/network/io-service;1"]
+                    .getService(Ci.nsIIOService);
+
+          let uri = ios.newURI(url, null, null);
+          let results = bs.getBookmarkIdsForURI(uri, {});
+
+          return results.length == 1;
+        """, script_args=[url])
+
+    def get_folder_ids_for_url(self, url):
+        """Retrieves the folder ids where the given URL has been bookmarked in.
+
+         :param url: URL of the bookmark
+
+         :returns: List of folder ids
```

Comment:
All the other libraries also have this style, so I just followed it. Personally I find it better with separation, but we can also globally remove that again. I'm open to that.